src/index.ts

```
@@ -1159,6 +1162,19 @@ class Tinypool extends EventEmitterAsyncResource {
       )
     }
 
+    if (isBun) {
+      if (options.useAtomics) {
+        throw new Error('options.useAtomics can not be set in Bun runtime')
+      }
+
+      // ::bunternal:: [NotImplementedError]: worker_threads.Worker option "resourceLimits" is not yet implemented in Bun.
+      if (options.resourceLimits) {
+        throw new Error('options.resourceLimits can not be set in Bun runtime.')
+      }
+
+      options.useAtomics = false
+    }
+
```

I would rather remove this `if (isBun)` branch completely so we don't need to track Bun's support. Is the error message from Bun descriptive in these cases?

---

For `useAtomics` case it just hangs without error, so we have to keep it anyway. For `resourceLimits` it throws a long error trace during execution. So I kept this check during initialization to fail early than execution time. 

```
::bunternal:: [NotImplementedError]: worker_threads.Worker option "resourceLimits" is not yet implemented in Bun.
    at new ::bunternal:: (internal:shared:7:4)
    at warnNotImplementedOnce (internal:shared:26:36)
    at new Worker (node:worker_threads:147:30)
    at initialize (/Users/nazar/Hub/Lodestar/Projects/tinypool/dist/index.js:67:23)
    at _addNewWorker (/Users/nazar/Hub/Lodestar/Projects/tinypool/dist/index.js:534:9)
    at _ensureMinimumWorkers (/Users/nazar/Hub/Lodestar/Projects/tinypool/dist/index.js:521:22)
    at new ThreadPool (/Users/nazar/Hub/Lodestar/Projects/tinypool/dist/index.js:512:15)
    at new Tinypool (/Users/nazar/Hub/Lodestar/Projects/tinypool/dist/index.js:713:4)
    at /Users/nazar/Hub/Lodestar/Projects/tinypool/test/resource-limits.test.ts:11:9
    at /Users/nazar/Hub/Lodestar/Projects/tinypool/test/resource-limits.test.ts:8:46
```